c2rust-analyze/src/util.rs

```diff
@@ -177,7 +187,68 @@ fn builtin_callee<'tcx>(
             Some(Callee::MiscBuiltin)
         }
 
-        _ => None,
+        "size_of" => {
+            // `core::mem::size_of`
+            let path = tcx.def_path(did);
+            if tcx.crate_name(path.krate).as_str() != "core" {
+                return None;
+            }
+            if path.data.len() != 2 {
+                return None;
+            }
+            if path.data[0].to_string() != "mem" {
+                return None;
+            }
+            Some(Callee::MiscBuiltin)
+        }
+
+        "malloc" | "c2rust_test_typed_malloc" => {
+            if matches!(tcx.def_kind(tcx.parent(did)), DefKind::ForeignMod) {
+                return Some(Callee::Malloc);
+            }
+            None
+        }
+
+        "calloc" | "c2rust_test_typed_calloc" => {
+            if matches!(tcx.def_kind(tcx.parent(did)), DefKind::ForeignMod) {
+                return Some(Callee::Calloc);
+            }
+            None
+        }
+
+        "realloc" | "c2rust_test_typed_realloc" => {
+            if matches!(tcx.def_kind(tcx.parent(did)), DefKind::ForeignMod) {
+                return Some(Callee::Realloc);
+            }
+            None
+        }
+
+        "free" | "c2rust_test_typed_free" => {
+            if matches!(tcx.def_kind(tcx.parent(did)), DefKind::ForeignMod) {
+                return Some(Callee::Free);
+            }
+            None
+        }
+
+        "is_null" => {
+            // `core::ptr::is_null`
```

Comment:
This case should use the same logic as for `offset`, checking that the function is an inherent method of a raw pointer type.  The actual path of `is_null` is not `core::ptr::is_null` but rather something like `core::ptr::const_ptr::{{impl}}::is_null`.  Checking the parent impl's self type is more precise and robust against standard library changes.

Reply:
changed in 197ffe22